src/main/java/software/amazon/event/kafkaconnector/EventBridgeWriter.java

```diff
@@ -120,54 +128,90 @@ public EventBridgeWriter(EventBridgeAsyncClient ebClient, EventBridgeSinkConfig
    * @param records
    * @return list of all records with additional status information
    */
-  public List<EventBridgeWriterRecord> putItems(List<EventBridgeWriterRecord> records) {
-    try {
-      var putEventsEntries =
-          records.stream()
-              .map(this::createPutEventsEntry)
-              .flatMap(Optional::stream)
-              .collect(Collectors.toList());
+  public List<EventBridgeWriterResult> putItems(List<SinkRecord> records) {
+    return doWithPutEventsRequestFrom(
```

Comment:
Can you elaborate why you went with this approach? Is it to avoid the vars outside the try-catch block? I really like that we are splitting it in smaller parts - but I'm not sure if passing the Lambdas makes it more concise. Thinking of the upcoming batching changes I see this getting complex and one has to dive deeper to understand. Maybe let's compare with your initial "sequential" design and decide which way to go? Can you add add it here for comparison?

Reply:
The reason is to extract the functionality of mapping a list of `SinkRecord`s to the required `PutEventsRequest`. Like a class should be have a single responsibility, it's a good practice to not overwhelm a function/method.

It might not be a lambda it can also be a method reference. Or even better, the mapping should be extracted to a dedicated interface and implementation. This come in handy for #39.

Reply:
Yes I think we are on the same page here with splitting it into smaller parts. What I wasn't sure about was the BiFunction way of handling it. What I meant was more like:

```
  public List<EventBridgeWriterResult> putItems(List<SinkRecord> records) {
      var mappedRecords = getMappedRecords(records);
      var success = x; //Get success mapped records here
      var errors = y; //Get error mapped records here

      if (success.isEmpty()){
        return errors;
      }
      //Later we can easily batch/loop this here and apply the new collector when setting "success"
      var results = sendToEB(success);

      return concat(results, errors);
  }
```